gc3libs/config.py

```diff
@@ -438,11 +447,18 @@ def _parse(self, stream, filename=None):
         'type',
     )
 
+    # config keys that are supposed to have a pattern
+    resource_key_value_matcher = {
+        'keypair_name': (r'/^$|\s+/', 'cannot be empty'),
+        'public_key': (r'\w*.pub\b', 'public keys need to end in .pub')
```

Comment:
This RE for `public_key` will disallow use of `/` or  dots or `~` in the key name; e.g. a value like `~/id_rsa.2017.pub` would be disallowed -- and it shouldn't!

If we want to enforce that public keys and with `.pub` (which is *not* exactly what #572 is about), then a better strategy IMHO would be to: use `re.search()` instead of `re.match()` and change this regexpp to just check that the ending is `.pub` (e.g., `r'.pub$'`)